h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostScoreTask.java

```diff
@@ -137,11 +142,20 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
             // This might be fixed in future versions of XGBoost
             Rabit.init(rabitEnv);
 
+            final int weightColumnIndex;
+            DataInfo dataInfo = _sharedmodel._dataInfoKey.get();
+            if (dataInfo._weights) {
+                weightColumnIndex = dataInfo.weightChunkId();
+            } else {
+                weightColumnIndex = -1;
+            }
+
+            _sharedmodel._dataInfoKey.get().weightChunkId();
             data = XGBoostUtils.convertChunksToDMatrix(
                     _sharedmodel._dataInfoKey,
                     cs,
                     _fr.find(_parms._response_column),
-                    -1, // not used for preds
```

Comment:
Why is this wrong? You don't need weights to make predictions. It is not necessary to make predictions for rows with zero weight but it doesn't matter in terms of correctness. What matters is that the weights are accounted for in the MetricBuilder - which is true - MB is skipping predictions with zero weights.